src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+With this secrets engine, services can get certificates without going through the usual manual process of generating a private key and CSR, submitting to a CA, and waiting for a verification and signing process to complete.
+Vault's built-in authentication and authorization mechanisms provide the verification functionality.
+
+Spring Vault supports issuing, signing, revoking certificates, and CRL retrieval through `VaultPkiOperations`.
+All other PKI functionality can be used through `VaultOperations`.
+
+The following examples explain briefly the use of how to issue and revoke certificates:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/PKI.java[tags=pkiApi]
+----
+<1> Construct a certificate request using the `VaultCertificateRequest` builder.
+<2> Request a certificate from Vault.
+Vault acts as certificate authority and responds with a signed X.509 certificate.
+The actual response is a `CertificateBundle`.
+<3> Generated certificates can be loaded directly into a Java `KeyStore` containing public and private keys as well as the issuer certificate. `KeyStore` has a wide range of uses which makes this format suitable to configure for example a HTTP client, a database driver or a SSL-secured HTTP server.
+<4> `CertificateBundle` allows accessing the private key, public and issuer certificate directly through the Java Cryptography Extension API.
+<5> Once a certificate is no longer in use, it can be revokey through its serial number.
+Vault will include the revoked certificate in its CRL.
+====
+
+You can find more details about the https://www.vaultproject.io/api-docs/secret/pki[Vault PKI secrets API] in the Vault reference documentation.
+
+[[vault.core.backends.token]]
+== Token Authentication Backend
+
+This backend is an authentication backend that doesn't interact with actual secrets.
+Rather, it gives access to access token management.
+You can read more about <<vault.authentication.token,Token-based authentication>> in the <<vault.core.authentication, authentication methods chapter>>.
+
+The `token` authentication method is built-in and automatically available at `/auth/token`.
+It allows users to authenticate using a token, as well to create new tokens, revoke secrets by token, and more.
+
+When any other auth method returns an identity, Vault core invokes the token method to create a new unique token for that identity.
+
+The token store can also be used to bypass any other auth method: you can create tokens directly, as well as perform a variety of other operations on tokens such as renewal and revocation.
+
+Spring Vault uses this backend to renew and revoke session tokens supplied by the configured <<vault.core.authentication, authentication method>>.
+
+The following examples shows how to request, renew and revoke a Vault token from within your application:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Token.java[tags=tokenApi]
+----
+<1> Create an token applying role defaults.
```

Comment:
Change to "<1> Create a token by applying role defaults."

("by" matters here, because, without "by", the sentence can be understood as "Create a token that applies role defaults." when the truth is that applying the defaults creates the token. It's a tricky subtlety.)